api/scripts/certification/fix-competence-marks.js

```diff
@@ -2,35 +2,25 @@ import 'dotenv/config';
 
 import { knex } from '../../db/knex-database-connection.js';
 import { SESSIONS_VERSIONS } from '../../src/certification/shared/domain/models/SessionVersion.js';
+import { DelayMixin } from '../../src/shared/application/scripts/addons/delay.js';
+import { DryRunMixin } from '../../src/shared/application/scripts/addons/dryRun.js';
 import { Script } from '../../src/shared/application/scripts/script.js';
 import { ScriptRunner } from '../../src/shared/application/scripts/script-runner.js';
 import { PIX_ORIGIN } from '../../src/shared/domain/constants.js';
 import { Assessment } from '../../src/shared/domain/models/Assessment.js';
 
-export class FixCompetenceMarksScript extends Script {
+export class FixCompetenceMarksScript extends Script.withAddons(DelayMixin, DryRunMixin) {
```

Comment:
J'aime beaucoup la proposition sur la notion d'addons de scripts ❤️

[suggestion] Au niveau du design, j'aurais plutôt tendance à décrire les addons au niveau du constructeur:
```suggestion
export class FixCompetenceMarksScript extends Script {
  constructor() {
    super({
      description: 'Fix "competenceId" in table "competence-marks" from bad referential origin',
      permanent: false,
      addons: [DelayAddon, DryRunAddon],
    })
```

Ou bien en propriété de la classe (je ne sais pas si c'est possible):
```suggestion
export class FixCompetenceMarksScript extends Script {
  addons = [DelayAddon, DryRunAddon];
      
  constructor() {
    super({
      description: 'Fix "competenceId" in table "competence-marks" from bad referential origin',
      permanent: false,
    })
```

Reply:
L'qapproche 1 par constructor ne me semble pas applicable car on est deja trop loin dans la création de l'objet pour ce que l'on veut faire.

L'option 2 en propriété de classe me plait +, cependant ça m'évoque + une approche type Vue.js ([Composition API](https://fr.vuejs.org/guide/extras/composition-api-faq)).
Mais je suis pas certain que on veuille créer une mécanique de lifecycle hook pour nos script. 💭 

J'aimerais bien prendre l'option 2 mais j'ai l'impression que l'on va perdre la simplicité et les forces de la mécanique de [mixin](https://justinfagnani.com/2015/12/21/real-mixins-with-javascript-classes/#bettermixinsthroughclassexpressions) (paragraphe "Better Mixins Through Class Expressions")

Je veux bien explorer ensemble l'option 2 @bpetetot si tu vois une stratégie de mise en place 😊 

Si tu es un peu dérangé par l'écriture `Script.withAddons` derrière le extend (qui simplifie l'écriture des mixins ou tu aurais eu sinon `DelayAddon(DryRunAddon(Script))`) 

[suggestion] on peut eventuelement l'ecrire comme ça:
```suggestion
// import ...
const BuiltInScript = Script.withAddons(DelayMixin, DryRunMixin);

export class FixCompetenceMarksScript extends BuiltInScript {
  constructor() {
    super({
      description: 'Fix "competenceId" in table "competence-marks" from bad referential origin',
      permanent: false,
      addons: [DelayAddon, DryRunAddon],
    })
```

Ou alors on ne passe plus du tout par du `extend` et on cree un `ScriptBuilder` comme le `MixinBuilder` de l'article que j'ai partagé + haut.

Des gens chquds pour en discuter ? 😊

Reply:
Motivé pour en discuter (de vive-voix se sera plus facile). Mardi prochain au tech time ?